cookbooks/tian_parameterization_kinematic_slab/coupled_two_phase_tian_parameterization_kinematic_slab.prm

```diff
@@ -0,0 +1,234 @@
+                              onset_time=100e3, slab_thickness=20e3
+    set Function expression = initial_porosity; \
+                              if( t<=onset_time, 0, \
+                              if( (y<=slab_thickness - sediment_min) & (y>slab_thickness - sediment_max), initial_bound_sed, \
+                              if( (y<=slab_thickness - MORB_min) & (y>slab_thickness - MORB_max), initial_bound_MORB, \
+                              if( (y<=slab_thickness - gabbro_min) & (y>slab_thickness - gabbro_max), initial_bound_gabbro, 0.0)))); \
+                              if(y>slab_thickness, 1, 0); \
+                              if(y<=8e3, 1, 0); \
+                              if(y>8e3 & y<=15e3, 1, 0); \
+                              if(y>15e3 & y<=slab_thickness, 1, 0)
+  end
+end
+
+subsection Geometry model
+  set Model name = box
+  subsection Box
+    set X extent      = 100e3
+    set Y extent      = 100e3
+    set Y repetitions = 2
+    set X repetitions = 2
+  end
+end
+
+# Impose a gravity vector at an angle of 45 degrees. This
+# accounts for the fact that the slab is subducting with a dip
+# of 45 degrees and the x-axis of our model aligns parallel to the
+# slab surface.
+subsection Gravity model
+  set Model name = function
+  subsection Function
+    set Variable names      = x,y
+    set Function constants  = angle=0.7071067811865476
+    set Function expression = 9.81*angle; -9.81*angle
+  end
+end
+
+
+# Initial temperature is a constant value of 1500 K
+subsection Initial temperature model
+  set Model name = function
+  subsection Function
+    set Coordinate system   = cartesian
+    set Variable names      = x,y
+    set Function constants  = Tmantle=1500
+    set Function expression = Tmantle
+  end
+end
+
+# Fix the temperature on the left boundary to ensure that
+# the incoming slab is always colder than the mantle. The
+# temperature is unconstrained on other boundaries.
+subsection Boundary temperature model
+  set Fixed temperature boundary indicators   = left
+  set List of model names                     = function
+  subsection Function
+    set Coordinate system                     = cartesian
+    set Variable names                        = x,y
+    set Function constants                    = Tmantle=1500, Tslab=600, slab_thickness=20e3
+    set Function expression                   = if(y<=slab_thickness, Tslab - 200*y/slab_thickness, Tmantle)
+  end
+end
+
+
+# Use the reactive fluid transport model composited with the visco plastic material model.
+# We use the tian approximation method for the solid-fluid reactions, and place a maximum
+# wt% water on the 4 lithologies.
+subsection Material model
+  set Model name                                        = reactive fluid transport
+  set Material averaging                                = harmonic average only viscosity
+
+  subsection Reactive Fluid Transport Model
```

Comment:
It would be good here or in the cookbook to discuss the choices for these parameters (i.e., what physical observations and/or previous studies are they based on).